kotlinx-coroutines-core/common/src/JobSupport.kt

```diff
@@ -480,7 +484,14 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
                                 // or we are adding a child to a coroutine that is not completing yet
                                 if (rootCause == null || handler.isHandlerOf<ChildHandleNode>() && !state.isCompleting) {
```

Comment:
[nit] while we are here, please remove `isHandlerOf` function and replace it with `handler is ChildHandleNode`

Reply:
Kotlin/Js compilation will fail then with "Incompatible types" error.

Reply:
Then please drop a comment and report it to JS team

Reply:
Ok